lib/src/source_visitor.dart

```diff
@@ -1517,15 +1517,14 @@ class SourceVisitor extends ThrowingAstVisitor {
   }
 
   void visitIfElement(IfElement node) {
-    // Wrap the whole thing in a single rule. If a split happens inside the
-    // condition or the then clause, we want the then and else clauses to split.
-    builder.startRule();
-
-    token(node.ifKeyword);
-    space();
-    token(node.leftParenthesis);
-    visit(node.condition);
-    token(node.rightParenthesis);
+    // Treat a chain of if-else elements as a single unit so that we don't
+    // unnecessarily indent each subsequent section of the chain.
+    var ifElements = <IfElement>[];
```

Comment:
[optional]
```dart
    var ifElements = <IfElement>[
      for (CollectionElement thisNode = node;
          thisNode is IfElement;
          thisNode = (thisNode as IfElement).elseElement)
        thisNode,
    ];
```

Reply:
What is this bizarre syntax?!

...done.